gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDataset.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    // Adding this check to avoid adding post publish step when there are no files to copy.
+    if (CollectionUtils.isNotEmpty(destDataFiles)) {
+      copyEntities.add(createOverwritePostPublishStep(destDataFiles));
+    }
+
+    log.info("~{}~ generated {} copy--entities", fileSet, copyEntities.size());
+    return copyEntities;
+  }
+
+  private List<DataFile> getDestDataFiles(List<DataFile> srcDataFiles) throws IcebergTable.TableNotFoundException {
+    List<DataFile> destDataFiles = new ArrayList<>();
+    if (srcDataFiles.isEmpty()) {
+      log.warn("No data files found for partition col : {} with partition value : {} in source table : {}",
+          this.partitionColumnName, this.partitionColValue, this.getFileSetId());
+      return destDataFiles;
+    }
+    TableMetadata srcTableMetadata = getSrcIcebergTable().accessTableMetadata();
+    TableMetadata destTableMetadata = getDestIcebergTable().accessTableMetadata();
+    PartitionSpec partitionSpec = destTableMetadata.spec();
+    String srcWriteDataLocation = srcTableMetadata.property(TableProperties.WRITE_DATA_LOCATION, "");
+    String destWriteDataLocation = destTableMetadata.property(TableProperties.WRITE_DATA_LOCATION, "");
+    if (StringUtils.isEmpty(srcWriteDataLocation) || StringUtils.isEmpty(destWriteDataLocation)) {
+      log.warn(
+          "Either source or destination table does not have write data location : source table write data location : {} , destination table write data location : {}",
+          srcWriteDataLocation,
+          destWriteDataLocation
+      );
+    }
+    // tableMetadata.property(TableProperties.WRITE_DATA_LOCATION, "") returns null if the property is not set and
+    // doesn't respect passed default value, so to avoid NPE in .replace() we are setting it to empty string.
+    String prefixToBeReplaced = (srcWriteDataLocation != null) ? srcWriteDataLocation : "";
+    String prefixToReplaceWith = (destWriteDataLocation != null) ? destWriteDataLocation : "";
+    GrowthMilestoneTracker growthMilestoneTracker = new GrowthMilestoneTracker();
+    srcDataFiles.forEach(dataFile -> {
+      String srcFilePath = dataFile.path().toString();
+      Path updatedDestFilePath = relocateDestPath(srcFilePath, prefixToBeReplaced, prefixToReplaceWith);
+      destDataFiles.add(DataFiles.builder(partitionSpec)
+          .copy(dataFile)
+          .withPath(updatedDestFilePath.toString())
+          .build());
+      // Store the mapping of srcPath to destPath to be used in creating list of src file status to dest path
+      log.debug("Path changed from Src : {} to Dest : {}", srcFilePath, updatedDestFilePath);
+      srcPathToDestPath.put(new Path(srcFilePath), updatedDestFilePath);
+      if (growthMilestoneTracker.isAnotherMilestone(destDataFiles.size())) {
+        log.info("Generated {} destination data files", destDataFiles.size());
+      }
+    });
+    log.info("Generated {} destination data files", destDataFiles.size());
```

Comment:
as mentioned earlier, let's keep the logging convention (that names the table) from `IcebergDataset`, specifically:
```
log.info("~{}~ doing ...", this.getFileSetId() ...)
```